helm: renamed podsAnnotations variable into podAnnotations

mvisonneau opened the pull request:
This change follows-up onto https://github.com/cilium/cilium/pull/12189#issuecomment-705226944
It concerns the agent and operator charts

sayboras commented on install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:
```diff
@@ -27,7 +27,7 @@ spec:
         # https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
         scheduler.alpha.kubernetes.io/critical-pod: ""
-{{- with .Values.podsAnnotations }}{{- toYaml . | nindent 8 }}{{- end }}
+{{- with .Values.podAnnotations }}{{- toYaml . | nindent 8 }}{{- end }}
```
Need other to weigh in, as I am not sure how we capture note for upgrad for this changes. Using `--reuse-values` during upgrade might remove existing pod annotations if this field is not explicitly set in `helm upgrade`

sayboras replied:
the above is answered by Chris :)

christarazi reviewed (approved):
Thanks! We'll need to wait until the big Helm refactor is merged.

@sayboras I think because the original change never made it into a release, we could probably skip mentioning it. It seems like this PR is just amending the previous change. (Unless I've missed something.)

sayboras reviewed (approved):
LGTM 💯

kaworu reviewed (changes requested):
Here is my understanding:

https://github.com/cilium/cilium/pull/13259 does include this fix, so the issue will be solved as soon as #13259 is merged into master.

There is a need to have this fix in v1.8. Since this PR won't be merged into master, it could be re-targeted to be merged into the v1.8 branch. Alternatively, this PR could be closed and another PR could be opened against v1.8.

mvisonneau committed:
helm: configurable annotations for agent and operator pods

Signed-off-by: Maxime VISONNEAU <maxime.visonneau@gmail.com>

mvisonneau committed:
helm: renamed podsAnnotations variable into podAnnotations

This change follows-up onto https://github.com/cilium/cilium/pull/12189#issuecomment-705226944

Signed-off-by: Maxime VISONNEAU <maxime.visonneau@gmail.com>

kaworu reviewed (approved):
LGTM, thanks!
